docs/src/proposals/secp256k1_instruction.md

```diff
@@ -0,0 +1,75 @@
+---
+title: secp256k1 builtin instruction
+---
+
+## Problem
+
+Need to efficiently perform multiple secp256k1 pubkey recovery operations (ecrecover) in a transaction.
+ecrecover is an ethereum instruction which takes a signature and message and recovers a publickey, a comparison
+to that public key can thus verify that the signature is valid.
+
+The input signature and message data can come in the transaction itself, but there is a requirement for the pubkeys
```

Comment:
Where does this requirement come from exactly? Can we simulate the transaction to extract the public key and then put it into the instruction?

Reply:
This is what Certus requires in their design. I'm not sure exactly how hard the requirement is. I'm not sure what "simulate the transaction" means. Do you mean execute the ecrecover function to get the public key and then pass it to the following instructions? That seems do-able.

Reply:
Removes the possibility of overlapping execution though.

Reply:
Not necessarily execute the whole transaction, though that'd work in the general cade. I'd be sufficient to query the account data beforehand, grab the pubkey, and use it to create the transaction.